river/render.zig

```diff
@@ -47,13 +47,65 @@ const SurfaceRenderData = struct {
     opacity: f32,
 };
 
+fn countSurfaceIterator(
+    surface: *wlr.Surface,
+    surface_x: c_int,
+    surface_y: c_int,
+    count: *u32,
+) callconv(.C) void {
+    count.* += 1;
+}
+
+fn scanOutView(output: *Output, view: *View) bool {
+    // Some things want to be rendered even above fullscreen surfaces. We need
+    // to composite in those cases.
+    if (build_options.xwayland and server.root.xwayland_unmanaged_views.next != null) return false;
+    if (output.getLayer(.overlay).first != null) return false;
+    if (server.root.drag_icons.first != null) return false;
+
+    if (view.saved_buffers.items.len != 0) return false;
+    if (view.surface == null or view.surface.?.buffer == null) return false;
+
+    // If the surface has subsurfaces we need to composite.
+    var surfaces: u32 = 0;
+    view.surface.?.forEachSurface(*u32, countSurfaceIterator, &surfaces);
+    if (surfaces > 1) return false;
```

Comment:
cant we just check if `surface.subsurfaces_{above,below}` are empty instead of iterating?

Reply:
Yes we can.